fedimint-core/src/task/jit.rs

```diff
@@ -110,27 +110,28 @@ where
             .val
             .get_or_init(|| async {
                 let handle: &mut _ = &mut *self.inner.handle.lock().await;
-                handle
-                    .await
-                    .unwrap_or_else(|err| {
-
-                        #[cfg(not(target_family = "wasm"))]
-                        if err.is_panic() {
-                            warn!(target: LOG_TASK, %err, type_name = %std::any::type_name::<T>(), "Jit value panicked");
-                            // Resume the panic on the main task
-                            panic::resume_unwind(err.into_panic());
-                        }
-                        #[cfg(not(target_family = "wasm"))]
-                        if err.is_cancelled() {
-                            warn!(target: LOG_TASK, %err, type_name = %std::any::type_name::<T>(), "Jit value task canceled:");
-                        }
-                        panic!("Jit value {} failed unexpectedly with: {}", std::any::type_name::<T>(), err);
```

Comment:
Took me way too long to find that diff :laughing: would have been nice if the pure refactoring was a separate PR, or at least if there was a comment pointing to the actual change :see_no_evil:

Reply:
Oh. The whole outside thing had to change due needing to return some value. So there's no smaller way that would compile (that I could come up with).